contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,821 @@
+
+ - `kubectl rollout status statefulset <StatefulSet-Name>`: displays the status 
+ of a StatefulSet update.
+ - `kubectl rollout undo statefulset <StatefulSet-Name>`: triggers a rollback 
+ of the current update.
+ - `kubectl rollout history statefulset <StatefulSet-Name>`: displays a the 
+ StatefulSets revision history.
+
+## Usage
+This section demonstrates how the design functions in typical usage scenarios.
+
+### Initial Deployment
+Users can create a StatefulSet using `kubectl apply`.
+
+Given the following manifest `web.yaml`
+
+```yaml
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: web
+spec:
+  serviceName: "nginx"
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: gcr.io/google_containers/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
+        volumeMounts:
+        - name: www
+          mountPath: /usr/share/nginx/html
+  volumeClaimTemplates:
+  - metadata:
+      name: www
+      annotations:
+        volume.alpha.kubernetes.io/storage-class: anything
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 1Gi
+```
+
+Users can use the following command to create the StatefulSet.
+
+```shell
+kubectl apply -f web.yaml
+```
+
+The only difference between the proposed and current implementation is that 
+the proposed implementation will initialize the StatefulSet's revision history 
+upon initial creation.
+
+### Rolling out an Update
+Users can create a rolling update using `kubectl apply`. If a user creates a 
+StatefulSet [as above](#initial-deployment), the user can trigger a rolling 
+update by updating the image (as in the manifest as below).
+
+```yaml
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: web
+spec:
+  serviceName: "nginx"
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      updateStrategy: 
+        type: RollingUpdate
+      containers:
+      - name: nginx
+        image: gcr.io/google_containers/nginx-slim:0.9
+        ports:
+        - containerPort: 80
+          name: web
+        volumeMounts:
+        - name: www
+          mountPath: /usr/share/nginx/html
+  volumeClaimTemplates:
+  - metadata:
+      name: www
+      annotations:
+        volume.alpha.kubernetes.io/storage-class: anything
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 1Gi
+```
+
+
+Users can use the following command to trigger a rolling update.
+
+```shell
+kubectl apply -f web.yaml
+```
+
+### Canaries
+Users can create a canary using `kubectl apply`. The only difference between a
+ [rolling update](#rolling-out-an-update) and a canary is that the 
+ `.Spec.UpdateStrategy.Type` is set to `ParitionedStatefulSetStrategyType` and 
+ the `.Spec.UpdateStrategy.Partition.Ordinal` is set to `.Spec.Replicas-1`.
```

Comment:
@kow3ns we should consider highlighting here how the different fields replicas,updated,current would differ in canary and normal case. Its confusing to the user